### DISCUSSION TITLE
Tracy: podpora pro vice loggeru

### DISCUSSION THREAD

**Discussion by f3l1x** [nette/nette#1557] [General]
## RFC

Balicek: tracy/tracy

## Usecase

V dnesni dobe je pomerne dobry zvyk pouzivat nejaky error tracking nastroj (napr. sentry), kam se zasilaji informace o exceptions v aplikaci. Lze to vyresit nekolika zpusoby, ale nabizi se celkem jednoduche reseni, ktere jsem aplikoval na nekolik projektu a to pridat tuto podporu do Tracy, resp. mit vice loggeru. Tracy totiz chytre trackuje vsechny exceptions a pripadne umi i notice konvertovat na exceptions.

## Implementation

Predstavuji si to velmi jednoduse a to tak, ze by Tracy\Debugger mel metodu addLogger.  Je totiz zadouci, aby fungoval i klasicky Tracy Logger, protoze do Sentry a dalsich nastroju nelze poslat vsechny informace, jake jsou obsazene v HTML ladence. Proto je tvorba HTML ladenek dobre zachovat.

```php
Debugger::addLogger(new SentryLogger($options));
```

**Comment by dg** [comment]
Ohledně Loggeru vím že tam jsou nějaké pull requesty. 

Btw proč to neotevres v Tracy?

**Comment by f3l1x** [comment]
Rad bych to prave nejdrive probral interne. Zjistil jaky na to mas nazor a pak pripadne pripravil i implementaci. Chci maximalne vyuzit svuj cas a ne se dohadovat s ostatnima. Je to spatny?

**Comment by dg** [comment]
Jde o tyto issues https://github.com/nette/tracy/issues?q=logger+is%3Aopen. Rád bych je uzavřel. Názor na to nemám žádný, protože to nepotřebuju/nepoužívám.

Jako další verzi bych chtěl vydat Tracy 3, kde půjdu do BC breaků v API, takže není potřeba držet přísně kompatibilitu a je možné to navrhnout odznova.

**Comment by f3l1x** [comment]
Tak jakmile bude vyvoj Tracy 3, tak rekni a ja pripravim ten logger.

**Comment by dg** [comment]
Už je, v masteru.

**Comment by milo** [comment]
Možnost více loggerů bych uvítal, ale tak, abych byl schopný se dostat k souboru s red screen.

Teď to řeším vlastním `ChainLogger implements ILogger`. Typicky obsahuje výchozí Tracy logger, který vygeneruje red screen soubor, následuje SyslogLogger a poslední logger notifikuje RSCS (Red Screen Collector Server) :)
